In [48]:
import load_data


In [49]:
# Run the load_sec8_contracts function
timeline = load_data.load_sec8_contracts(load_data.sec8_flatfile_paths)

Starting 'load_sec8_contracts'. 3 snapshots to load; this may take some time... (hint use ctrl+c to interrupt) 
  Starting data from '2011'...
    Loaded 'contracts'
    Loaded 'properties'
  Finished snapshot '2011' from 2011-09-22. (19.53 seconds)
  Starting data from '2015'...
    Loaded 'contracts'
    Loaded 'properties'
  Finished snapshot '2015' from 2015-09-05. (21.83 seconds)
  Starting data from 'current'...
    Loaded 'contracts'
    Loaded 'properties'
  Finished snapshot 'current' from 2016-08-02. (10.2 seconds)
Load finished.
Total load time: 51.5592
------
Total snapshots loaded: 3


In [50]:
print(timeline)

In [57]:
cdf=[] #convenience variable

for i in range(0,3):    
    cdf.append(timeline.snapshots[i].contracts_df)
    print("contracts in " + str(i) + ":",cdf[i].property_id.count())

print(list(cdf[0].columns.values))
#cdf[0].describe()
cdf[0].sort_index(inplace=True)
cdf[1].sort_index(inplace=True)

contracts in 0: 23595
contracts in 1: 23507
contracts in 2: 23640
['contract_number', 'property_id', 'property_name_text', 'tracs_effective_date', 'tracs_overall_expiration_date', 'tracs_overall_exp_fiscal_year', 'tracs_overall_expire_quarter', 'tracs_current_expiration_date', 'tracs_status_name', 'contract_term_months_qty', 'assisted_units_count', 'is_hud_administered_ind', 'is_acc_old_ind', 'is_acc_performance_based_ind', 'contract_doc_type_code', 'program_type_name', 'program_type_group_code', 'program_type_group_name', 'rent_to_FMR_ratio', 'rent_to_FMR_description', '0BR_count', '1BR_count', '2BR_count', '3BR_count', '4BR_count', '5plusBR_count', '0BR_FMR', '1BR_FMR', '2BR_FMR', '3BR_FMR', '4BR_FMR']


I pulled the contracts for 2011 and 2015 (i.e. cdf[0] and cdf[1]) into an Excel file and did a comparison of 'tracs_overall_expiration_date'. I got these numbers:

Total:	                            23,507
New buildings in 2015:	               691
Deadline sooner in 2015 than 2011:	11,214
Deadline later in 2015 than 2011:	11,072
No change:	                           530

Not sure why this doesn't match with the 1,795 records for inner join below (should be the same as 'no change' field)


In [65]:
import pandas as pd
pd.merge(cdf[0],cdf[1],on=['contract_number','tracs_current_expiration_date'],how='inner')



contract_number  property_id_x  \
0        FL29Q921001      800000013   
1        VA36S941007      800000019   
2        CO99S931003      800000025   
3        NY06S921015      800000026   
4        MD39CS94002      800000034   
5        OH16Q941006      800000038   
6        IL06S921007      800000046   
7        AZ20S941009      800000047   
8        MA06Q901003      800000048   
9        OH16S921005      800000066   
10       CA16S901007      800000067   
11       WA19S911004      800000070   
12       CA33S911001      800000071   
13       LA48S931007      800000074   
14       AK020102002      800000088   
15       AK06Q941001      800000094   
16       AK020102005      800000098   
17       AK020102011      800000115   
18       AL09Q921010      800000127   
19       AL09S941001      800000130   
20       AL09S931003      800000157   
21       AL09Q921007      800000180   
22       AL09Q921003      800000187   
23       AL09Q931002      800000204   
24       AL09S941002      800000212   
25       AL09Q911002      800000224   
26       AL09S921005      800000248   
27       AL09S931004      800000250   
28       AL09Q911001      800000258   
29       AL09Q901003      800000276   
...              ...            ...   
1765     ND990014001      800212595   
1766     CO99Q001001      800213150   
1767     CA16Q001011      800213233   
1768     MA06Q011010      800215463   
1769     TN40Q011001      800215474   
1770     CA16S931007      800215601   
1771     MO36H195080      800216919   
1772     MO16M000173      800218414   
1773     071023NISUP      800219195   
1774     071025NISUP      800219234   
1775     GA06S031006      800219566   
1776     MA06E000100      800221137   
1777     RQ46S041001      800221437   
1778     MA06Q041003      800221510   
1779     TN40H112012      800222240   
1780     NE26Q051001      800222545   
1781     CO99Q051001      800222637   
1782     NY36S051001      800222669   
1783     RQ46S051001      800222698   
1784     NY06S051008      800223023   
1785     VA36S051004      800223152   
1786     RQ46Q051002      800223153   
1787     02319ANISUP      800223648   
1788     CAPF0019511      800224483   
1789     CAPF0019510      800224595   
1790     NC19S071001      800224779   
1791     MO16CS97001      800225295   
1792     MO16CS97002      800225295   
1793     071031NIRAP      800232839   
1794     IL06S051006      800234650   

                                   property_name_text_x  \
0     BRADENTON VOA LIVING CENTER                   ...   
1     Christian Village of Western Virginia         ...   
2     MONTROSE VOA                                  ...   
3     SHENENDEHOWA VILLAGE                          ...   
4     GATEWAY VILLAGE                               ...   
5     HILL MANOR, II                                ...   
6     MATTHEW MANOR                                 ...   
7     Christian Care Manor IV                       ...   
8     MHPI VII                                      ...   
9     SETON COSHOCTON                               ...   
10    South Bay Retirement Residence                ...   
11    MABLE SWAN NORTH                              ...   
12    TELACU Southbay Manor                         ...   
13    VILLAGE du VIE                                ...   
14    CHUGACH VIEW                                  ...   
15    FIREWEED HOUSING DEVELOPMENT                  ...   
16    GOLDEN TOWERS APARTMENTS                      ...   
17    PTARMIGAN PARK APARTMENTS                     ...   
18    AGAPE HOUSE                                   ...   
19    AHEPA 23-II APARTMENTS                        ...   
20    BAY MINETTE ROTARY VIL                        ...   
21    C.A.P.E.                                      ...   
22    THE ARC HOME                                  ...   
23    COMMUNITY CONCEPTS                            ...   
24    CREPE MYRTLE VILLAGE                          ...   
25    DECATUR VOA LIVING CENTER                     